tests/unit/tensor.cpp

```diff
@@ -0,0 +1,2214 @@
+    vkt::Tensor tensor(*m_device, info);
+    m_errorMonitor->VerifyFound();
+}
+
+TEST_F(NegativeTensor, StridesNotMultiple) {
+    TEST_DESCRIPTION("Test creating a tensor where the strides are not multiples of the size of tensor element in bytes");
+    SetTargetApiVersion(VK_API_VERSION_1_4);
+    AddRequiredExtensions(VK_ARM_TENSORS_EXTENSION_NAME);
+    AddRequiredFeature(vkt::Feature::tensors);
+    RETURN_IF_SKIP(Init());
+
+    VkTensorDescriptionARM desc = default_desc();
+    VkTensorCreateInfoARM info = default_create_info(&desc);
+    static std::vector<int64_t> dimensions{4ul, 1ul};
+    static std::vector<int64_t> strides{3l, 2l};
+    desc.pDimensions = dimensions.data();
+    desc.dimensionCount = dimensions.size();
+    desc.format = VK_FORMAT_R16_UINT;
+    desc.pStrides = strides.data();
+
+    m_errorMonitor->SetDesiredError("VUID-VkTensorDescriptionARM-pStrides-09737");
+    vkt::Tensor tensor(*m_device, info);
+    m_errorMonitor->VerifyFound();
+}
+
+TEST_F(NegativeTensor, StridesExtremes) {
+    TEST_DESCRIPTION("Test creating a tensor where the strides are less than or equal to zero, or greater than maxTensorStride");
+    SetTargetApiVersion(VK_API_VERSION_1_4);
+    AddRequiredExtensions(VK_ARM_TENSORS_EXTENSION_NAME);
+    AddRequiredFeature(vkt::Feature::tensors);
+    RETURN_IF_SKIP(Init());
+
+    VkTensorDescriptionARM desc = default_desc();
+    VkTensorCreateInfoARM info = default_create_info(&desc);
+
+    VkPhysicalDeviceTensorPropertiesARM tensor_props = vku::InitStructHelper();
+    GetPhysicalDeviceProperties2(tensor_props);
+
+    static std::vector<int64_t> dimensions{2ul, 2ul, 2ul, 2ul};
+    static std::vector<int64_t> strides{-1l, 0l, static_cast<int64_t>(tensor_props.maxTensorStride + 1), 1l};
+    desc.pDimensions = dimensions.data();
+    desc.dimensionCount = dimensions.size();
+    desc.pStrides = strides.data();
+
+    m_errorMonitor->SetDesiredError("VUID-VkTensorDescriptionARM-pStrides-09738");
+    m_errorMonitor->SetDesiredError("VUID-VkTensorDescriptionARM-pStrides-09738");
+    m_errorMonitor->SetDesiredError("VUID-VkTensorDescriptionARM-pStrides-09738");
+
+    // This test also triggers VUID-VkTensorDescriptionARM-pStrides-09739
+    m_errorMonitor->SetDesiredError("VUID-VkTensorDescriptionARM-pStrides-09739");
+    m_errorMonitor->SetDesiredError("VUID-VkTensorDescriptionARM-pStrides-09739");
+    vkt::Tensor tensor(*m_device, info);
+    m_errorMonitor->VerifyFound();
+}
+
+TEST_F(NegativeTensor, StridesOverlap) {
+    TEST_DESCRIPTION("Test creating a tensor where the strides overlap");
+    SetTargetApiVersion(VK_API_VERSION_1_4);
+    AddRequiredExtensions(VK_ARM_TENSORS_EXTENSION_NAME);
+    AddRequiredFeature(vkt::Feature::tensors);
+    RETURN_IF_SKIP(Init());
+
+    VkTensorDescriptionARM desc = default_desc();
+    VkTensorCreateInfoARM info = default_create_info(&desc);
+
+    static std::vector<int64_t> dimensions{1ul, 1ul, 1ul};
```

Comment:
why are of these static? can you just make this 

```suggestion
    std::array<int64_t> dimensions{1ul, 1ul, 1ul};
```

Reply:
yes, why static? It should be const.